File: packages/core/src/schemas/commit.ts
import { z } from "zod";
import { CardIdSchema } from "./card.js";
import { ExecutionIdSchema } from "./execution.js";

export const CommitIdSchema = z.string().brand<"CommitId">();

export const CardCommitSchema = z.object({
  id: CommitIdSchema,
  card_id: CardIdSchema,
  execution_id: ExecutionIdSchema.nullable(),
  sha: z.string(),
  message: z.string().default(""),
  author_name: z.string().default(""),
  author_email: z.string().default(""),
  files_changed: z.string().nullable().default(null),
  created_at: z.string(),
});

File: rust/crates/core/src/types/commit.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export, export_to = "../../../../packages/dashboard/src/types/generated/")]
pub struct CardCommit {
    pub id: String,
    pub card_id: String,
    pub execution_id: Option<String>,
    pub sha: String,
    pub message: String,
    pub author_name: String,
    pub author_email: String,
    pub files_changed: Option<String>,
    pub created_at: String,
}
